content/golang/docker/publish-images-multi-arch.md

```diff
@@ -15,45 +15,512 @@ tags:
+FROM alpine
+
+WORKDIR /app
+
+COPY hello-world /app
+
+EXPOSE 8080
 
-Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+ENTRYPOINT [ "/app/hello-world" ]
+{{</highlight>}}
 
-# Third Step
+Finally we build it with using:
+
+```
+docker build -t hello-world .
+```
+
+You can test the image with the following command:
+
+```
+docker run -it -p 8080:8080 hello-world
+```
+
+Then you should be able to see the hello-world application in your [browser](http://localhost:8080).
+
+# Continuous Integration and Delivery
+
+Even though the steps above are easy, it makes sense to automate them. 
+
+In order to do that, you will need to set up a [Git](https://git-scm.com) repository. For the rest of the tutorial we will assume, that the project is already on [GitHub](https://github.com/).
+
+In this section we configure [Drone](https://drone.io), an Open Source continuous integration system, to automatically build and test our code every time we push to GitHub. You can install Drone on your [own servers](https://docs.drone.io/installation/overview/), or you can use the [free Cloud offering](https://cloud.drone.io/).
+
+## Activation
+
+Fist, login into Drone and activate your repository:
+
+![Activate Repository](/screenshots/drone-repo-activate.png)
+
+## Configuration
+
+Next, define a [Continuous Integration Pipeline](https://docs.drone.io/configure/pipeline/) for your project. Drone looks for special `.drone.yml` file within repositories for the Pipeline definition.
+
+We will achieve the same result we did in [Our Docker build steps]({{< ref "#building-with-docker" >}}), but this time automatically.
+
+First we need to set some basic information for Drone about our pipeline:
+
+- The kind of the `.yml` file. In most cases this will be `pipeline`.
+
+- The type of our pipeline. We want to use Docker containers for each step.
+
+- The name of the pipeline. This can be anything you like.
+
+- We also set the platform we want to execute the pipeline on. This depends on what platform your [Drone Runner](https://docs.drone.io/installation/runners/docker/) is on.
+
+Then translate our steps [from earlier]({{< ref "#building-with-docker-for-arm64" >}}) into Drone Pipeline steps.
+
+First we instruct Drone to build our Go application with the specified environment variables.
+
+Then in an another step Drone will build the image with our `Dockerfile` and then push it to Docker hub using the official [Drone Docker plugin](http://plugins.drone.io/drone-plugins/drone-docker/).
+
+Our final Pipeline will look like this:
+
+{{<highlight text "linenos=table,hl_lines=99">}}
+kind: pipeline
+type: docker
+name: build
+
+platform:
+  os: linux
+  arch: arm64
+
+steps:
+  - name: build
+    image: golang:1.13
+    environment:
+      CGO_ENABLED: "0"
+      GOOS: "linux"
```

Comment:
You can drop GOOS and GOARCH as these variables are automatically set by the running architecture.

Reply:
Same as with the docker image arch, they're there in order for the reader to know why they're needed and what they're for, but added notes if they can be omitted (and also removed them from the later parts of the tutorial).